src/Acuminator/Acuminator.Tests/Tests/StaticAnalysis/SavingChanges/SavingChangesInEventHandlersNonISVTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Acuminator.Analyzers.StaticAnalysis;
+using Acuminator.Analyzers.StaticAnalysis.EventHandlers;
+using Acuminator.Analyzers.StaticAnalysis.SavingChanges;
+using Acuminator.Tests.Helpers;
+using Acuminator.Tests.Verification;
+using Acuminator.Utilities;
+using Microsoft.CodeAnalysis.Diagnostics;
+using PX.Objects.FA;
+using Xunit;
+
+namespace Acuminator.Tests.Tests.StaticAnalysis.SavingChanges
+{
+	public class SavingChangesInEventHandlersNonISVTests : DiagnosticVerifier
+	{
+		protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer() =>
+			new EventHandlerAnalyzer(CodeAnalysisSettings.Default
+					.WithRecursiveAnalysisEnabled()
```

Comment:
The recursive analysis is enabled and ISV is disabled by default

Reply:
@saloponov I can't find your commentaboud code analysis settings in tests so I'll answer here.
We already have the ISV setting for a long time and it worked - the ISV is disabled by default. If there is something different returned from the `Default` property it needs to be fixed. Also notice that you don't specify `WithStaticAnalysisEnabled` setting and rely on default value.

Reply:
@SENya1990 we already used this approach and specified the settings that are important for the current test explicitly.
However, because `WithStaticAnalysisEnabled` was added recently, it was not used in tests.

Reply:
@dropsonic well, I checked out how many references we have to WIthRecursiveAnalysisEnabled, WithIsvSpecificAnalyzersEnabled and WithIsvSpecificAnalyzersDisabled. 
We have some WIthRecursiveAnalysisEnabled, I don't know why, because in .Net there are some default values for options and you need to know them (or specify a set). 
Because it kind of inconsistent in my opinion. In one test you specify them, in another you don't specify. 
And regarding the WithIsvSpecificAnalyzersDisabled -we have only one such call in dev and again I  most of the tests don't specify it.
But now I think I get what you're trying to do - you mark the test that specifically checks the  diagnostic messages when the ISV is turned off by calling explicitly these methods. I think it's ok but still not very consistent, maybe we need to discuss which tests should have such calls and update our code style

Reply:
Because it's unclear? Should we rewrite code analysis settings generation in all our tests every time we add a new option?

Reply:
@SENya1990 let's keep it "as is" and think about it later - it is out of the scope for this task

Reply:
@dropsonic, @saloponov, ok let's keep the tests as is, but I'll add a new section to our code guidelines that unit tests must always specify the effective set of code analysis parameters in order to be independent from the default values.
I'll also create a task to refactor old tests